tests/testthat/test_stan_betareg.R

```diff
@@ -0,0 +1,127 @@
+# of the License, or (at your option) any later version.
+# 
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+# 
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+# tests can be run using devtools::test() or manually by loading testthat 
+# package and then running the code below possibly with options(mc.cores = 4).
+
+library(rstanarm)
+library(betareg)
+SEED <- 12345
+set.seed(SEED)
+
+expect_stanreg <- function(x) expect_s3_class(x, "stanreg")
+
+link1 <- c("logit", "probit", "cloglog", "cauchit", "log", "loglog")
+link2 <- c("log", "identity", "sqrt")
+
+context("stan_betareg (dispersion only)")
+test_that("stan_betareg returns expected result when modeling x and dispersion", {
+  dat <- list()
+  dat$N <- 200
+  dat$x <- rnorm(dat$N, 2, 1)
+  dat$z <- rep(0, dat$N)
+  dat$mu <- binomial(link="logit")$linkinv(1+0.2*dat$x)
+  dat$phi <- 20
+  dat$y <- rbeta(dat$N, dat$mu * dat$phi, (1 - dat$mu) * dat$phi)
+  dat <- data.frame(dat$y, dat$x)
+  colnames(dat) <- c("y", "x")
+  for (i in 1:length(link1)) {
+    fit <- stan_betareg(y ~ x, link = link1[i], seed = SEED, QR = TRUE,
+                        prior = NULL, prior_intercept = NULL, prior_ops = NULL,
+                        data = dat, algorithm = "optimizing")
+    expect_stanreg(fit)
+    val <- coef(fit)
+    ans <- coef(betareg(y ~ x, link = link1[i], data = dat))
+    expect_equal(val, ans, tol = 0.1, info = link1[i])
+    cat("... used link = ", link1[i], "\n")
+  }
+})
+
+context("stan_betareg (x and z using link.phi = 'log')")
+test_that("stan_betareg returns expected result when modeling x and z", {
+  dat <- list()
+  dat$N <- 200
+  dat$x <- rnorm(dat$N, 2, 1)
+  dat$z <- rnorm(dat$N, 2, 1)
+  dat$mu <- binomial(link = "logit")$linkinv(0.5 + 0.2*dat$x)
+  dat$phi <- poisson(link = link2[1])$linkinv(1.5 + 0.4*dat$z)
+  dat$y <- rbeta(dat$N, dat$mu * dat$phi, (1 - dat$mu) * dat$phi)
+  dat <- data.frame(dat$y, dat$x, dat$z)
+  colnames(dat) <- c("y", "x", "z")
+  for (i in 1:length(link1)) {
+    cat("... using link =", link1[i], "and link.phi =", link2[1], "\n")
+    fit <- stan_betareg(y ~ x | z, link = link1[i], link.phi = link2[1], seed = SEED, QR = TRUE,
+                        prior = NULL, prior_intercept = NULL,
+                        prior_z = NULL, prior_intercept_z = NULL,
+                        data = dat, algorithm = "optimizing")
+    expect_stanreg(fit)
+    val <- coef(fit)
+    ans <- coef(betareg(y ~ x | z, link = link1[i], link.phi = link2[1], data = dat))
+    expect_equal(val, ans, tol = 0.1, info = c(link1[i], link2[1]))
+  }
+})
+
```

Comment:
If we are supporting `weights` and `offset` can you also add a quick test that just makes sure `stan_betareg` runs when `weights` and `offset` are specified and that the returned stanreg object has the correct  `weights` and `offset` values?